pkg/kubelet/cm/topologymanager/scope.go

```diff
@@ -69,9 +69,20 @@ func (s *scope) Name() string {
 }
 
 func (s *scope) GetAffinity(podUID string, containerName string) TopologyHint {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
 	return s.podTopologyHints[podUID][containerName]
 }
 
+func (s *scope) SetAffinity(podUID string, containerName string, hint TopologyHint) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	if s.podTopologyHints[podUID] == nil {
+		s.podTopologyHints[podUID] = make(map[string]TopologyHint)
+	}
+	s.podTopologyHints[podUID][containerName] = hint
+}
+
```

Comment:
The `GetAffinity()` function is part of the `Scope` interface (it is inherited from the `Store`interface).
As such, to introduce a `SetAffinity()` call here is a bit strange from the perspective of the `Scope` abstraction.
Consumers of this abstraction should not be able to change the affinity, only get it.

I'm fine adding a lock around access to this data structure, but we should do it in a way that doesn't extend / break the existing scope abstraction.

Reply:
@Klaven I think we should add `SetAffinity` into `Store`. It is so ugly if just lock in `Admit`.

Reply:
It is acceptable to create helper functions that do the locking and call them from within `GetAffinity()` and anywhere else you need. But as I said before, we should not introduce a `SetAffinity()` call to the interface.